src/charm.py

```diff
@@ -2342,6 +2349,25 @@ def _can_connect_to_postgresql(self) -> bool:
             return False
         return True
 
+    def _api_update_config(self) -> None:
+        # Use config value if set, calculate otherwise
+        max_connections = (
+            self.config.experimental_max_connections
+            if self.config.experimental_max_connections
+            else max(4 * self.cpu_count, 100)
+        )
+        cfg_patch = {
+            "max_connections": max_connections,
+            "max_prepared_transactions": self.config.memory_max_prepared_transactions,
+            "max_replication_slots": 25,
+            "max_wal_senders": 25,
+            "wal_keep_size": self.config.durability_wal_keep_size,
+        }
+        base_patch = {}
+        if primary_endpoint := self.async_replication.get_primary_cluster_endpoint():
+            base_patch["standby_cluster"] = {"host": primary_endpoint}
```

Comment:
Standby cluster is DCS value, so not updated by just changing the config. Should I also add this to the async relation changed hook?